contracts/BalancerV2Adapter.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: GPL-3.0
+
+/*
+    A Balancer V2 adapter for the Enzyme Protocol.
+*/
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@enzymefinance/contracts/release/extensions/integration-manager/integrations/utils/AdapterBase2.sol";
+
+/// @title BalancerV2Adapter Contract
+/// @author JustSomeGeeks Hackathon Team <https://github.com/justsomegeeks>
+/// @notice Adapter for interacting with Balancer (v2)
+/// @dev Does not allow any protocol that collects protocol fees in ETH, e.g., 0x v3
+contract BalancerV2Adapter is AdapterBase2 {
+    using SafeMath for uint256;
+
+    constructor(
+        address _integrationManager,
+        address _balancerSwapper,
+        address _tokenTransferProxy
+    ) public AdapterBase2(_integrationManager) {}
+
+    // EXTERNAL FUNCTIONS
+
+    /// @notice Provides a constant string identifier for an adapter
+    /// @return identifier_ An identifier string
+    function identifier() external pure override returns (string memory identifier_) {
+        return "BALANCER_V2";
+    }
+
+    /// @notice Parses the expected assets to receive from a call on integration
+    /// @param _selector The function selector for the callOnIntegration
+    /// @param _encodedCallArgs The encoded parameters for the callOnIntegration
+    /// @return spendAssetsHandleType_ A type that dictates how to handle granting
+    /// the adapter access to spend assets (`None` by default)
+    /// @return spendAssets_ The assets to spend in the call
+    /// @return spendAssetAmounts_ The max asset amounts to spend in the call
+    /// @return incomingAssets_ The assets to receive in the call
+    /// @return minIncomingAssetAmounts_ The min asset amounts to receive in the call
+    function parseAssetsForMethod(bytes4 _selector, bytes calldata _encodedCallArgs)
+        external
+        view
+        override
+        returns (
+            IIntegrationManager.SpendAssetsHandleType spendAssetsHandleType_,
+            address[] memory spendAssets_,
+            uint256[] memory spendAssetAmounts_,
+            address[] memory incomingAssets_,
+            uint256[] memory minIncomingAssetAmounts_
+        )
+    {
+        require(_selector == TAKE_ORDER_SELECTOR, "parseAssetsForMethod: _selector invalid");
+
+        // (
```

Comment:
Why are we leaving this comments in this function? I understand the others since they arent implemented yet and we need them for guidedance, but this seems implemented.

Reply:
this isn't implemented yet. i just used dummy variables to get things to compile:

eg:

```
spendAssets_ = new address[](1);
spendAssets_[0] = address(0x0);

spendAssetAmounts_ = new uint256[](1);
spendAssetAmounts_[0] = 0;

...
```

Those are all dummy variables until ` __decodeCallArgs(_encodedCallArgs)` is implemented.